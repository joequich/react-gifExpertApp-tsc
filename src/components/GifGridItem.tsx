import React from 'react'
import PropTypes from 'prop-types';

//IGifs
export interface Props {
    title: string;
    url: string;
}

const GifGridItem = ({title, url}:Props) => {
    return (
        <div role="listitem" className="card animate__animated animate__fadeInUp">
           <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};
GifGridItem.defaultProps = {
    id: ''
};

export default GifGridItem;