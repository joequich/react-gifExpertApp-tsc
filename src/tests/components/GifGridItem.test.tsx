import { render, screen } from '@testing-library/react';
import GifGridItem, { Props } from '../../components/GifGridItem';

describe('Test in <GifGridItem />', () => {

    const title = 'Hey';
    const url ='https//localhost/algo.png'
    let component = render(<GifGridItem title={title} url={url} />);   // spread operator

    // const defaultProps: Props = {
    //     title:'Hey',
    //     url: 'https//localhost/algo.png'
    // }
    
    // function renderUI(props: Partial<Props> = {}) {
    //     return render(<GifGridItem {...defaultProps} {...props}/>)
    // }

    beforeEach(() => {
        component = render(<GifGridItem title={title} url={url} />);   // spread operator
    });

    test('should display the component correctly', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });
    
    test('should have a paragraph with the title', () => {
        // render(<GifGridItem title={title} url={url} />); 
        expect(screen.getByText(title)).toHaveTextContent(title);
    });

    test('should have the same image as the url and the props image', () => {
        // render(<GifGridItem title={title} url={url} />);   // spread operator
        const img = screen.getByAltText(title);
        expect(img).toHaveAttribute('src', url);
        expect(img).toHaveAttribute('alt', title);
    });

    test('should have card animate__animated animate__fadeInUp', () => {
        // render(<GifGridItem title={title} url={url} />);   // spread operator
        const item = screen.getByRole('listitem');
        expect(item).toHaveClass('card animate__animated animate__fadeInUp', {exact: true});
        // expect(item.classList.contains('animate__fadeInUp')).toBe(true);
    });
});