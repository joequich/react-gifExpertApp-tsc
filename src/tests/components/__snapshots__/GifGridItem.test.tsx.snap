// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> should display the component correctly 1`] = `
<DocumentFragment>
  <div
    class="card animate__animated animate__fadeInUp"
    role="listitem"
  >
    <img
      alt="Hey"
      src="https//localhost/algo.png"
    />
    <p>
      Hey
    </p>
  </div>
</DocumentFragment>
`;
